WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt

```diff
@@ -298,12 +290,102 @@ class ReaderPostDetailViewModelTest {
+                    )
+                    assertThat(globalRelatedPosts).isEqualTo(
+                            RelatedPosts(posts = this@ReaderPostDetailViewModelTest.globalRelatedPosts, isGlobal = true)
+                    )
+                }
+            }
+
+    @Test
+    fun `given related posts fetch fails, when related posts are requested, then related posts are not shown`() =
+            test {
+                whenever(readerFetchRelatedPostsUseCase.fetchRelatedPosts(readerPost))
+                        .thenReturn(FetchRelatedPostsState.Failed.RequestFailed)
+                val uiStates = init().uiStates
+
+                viewModel.onRelatedPostsRequested(readerPost)
+
+                with(uiStates.last()) {
+                    assertThat(localRelatedPosts).isNull()
+                    assertThat(globalRelatedPosts).isNull()
+                }
+            }
+
+    @Test
+    fun `given no network, when related posts are requested, then related posts are not shown`() =
+            test {
+                whenever(readerFetchRelatedPostsUseCase.fetchRelatedPosts(readerPost))
+                        .thenReturn(FetchRelatedPostsState.Failed.NoNetwork)
+                val uiStates = init().uiStates
+
+                viewModel.onRelatedPostsRequested(readerPost)
+
+                with(uiStates.last()) {
+                    assertThat(localRelatedPosts).isNull()
+                    assertThat(globalRelatedPosts).isNull()
+                }
+            }
+
+    @Test
+    fun `given related posts fetch in progress, when related posts are requested, then related posts are not shown`() =
+            test {
+                whenever(readerFetchRelatedPostsUseCase.fetchRelatedPosts(readerPost))
+                        .thenReturn(FetchRelatedPostsState.AlreadyRunning)
+                val uiStates = init().uiStates
+
+                viewModel.onRelatedPostsRequested(readerPost)
+
+                with(uiStates.last()) {
+                    assertThat(localRelatedPosts).isNull()
+                    assertThat(globalRelatedPosts).isNull()
+                }
+            }
+
+    @Test
+    fun `given wp com post, when related posts are requested, then related posts are fetched`() =
+            test {
+                whenever(readerFetchRelatedPostsUseCase.fetchRelatedPosts(readerPost)).thenReturn(mock())
+
+                viewModel.onRelatedPostsRequested(readerPost)
+
+                verify(readerFetchRelatedPostsUseCase).fetchRelatedPosts(readerPost)
+            }
+
+    @Test
+    fun `given non wp com post, when related posts are requested, then related posts are not fetched`() =
+            test {
+                val nonWpComPost = createDummyReaderPost(id = 1, isWpComPost = false)
+
+                viewModel.onRelatedPostsRequested(nonWpComPost)
+
+                verify(readerFetchRelatedPostsUseCase, times(0)).fetchRelatedPosts(readerPost)
```

Comment:
Minor (🔍): I usually prefer the `never()` instead of `times(0)`. I am just mentioning that in case you weren't aware of it or you have a good reason I am not aware of using `times(0)` instead. Just sharing, nothing more, don't expect you to change it.